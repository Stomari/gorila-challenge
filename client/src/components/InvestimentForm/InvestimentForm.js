import React, { useState } from 'react';
import axios from 'axios';

const InvestimentForm = () => {
  const [type, setType] = useState('fixa');
  const [value, setValue] = useState('');
  const [date, setDate] = useState('');
  const [errorMessage, setErrorMessage] = useState(false);

  const submitHandler = (event) => {
    event.preventDefault();
    axios.post(`${process.env.REACT_APP_API_URL}/api/investiments`, { type, value, date })
      .then(() => {
        setType('fixa');
        setValue('');
        setDate('');
        setErrorMessage(false);
      })
      .catch((error) => {
        setErrorMessage(true)
        console.log(error);
      })
  }

  return (
    <section className="investiment-form">
      <form onSubmit={(event) => submitHandler(event)}>
        <label htmlFor="type">Tipo</label>
        <select name="type" value={type} onChange={(event) => setType(event.target.value)}>
          <option value="fixa">Renda Fixa</option>
          <option value="variavel">Renda Vari√°vel</option>
        </select>
        <label htmlFor="value">Valor</label>
        <input name="value" type="number" value={value} onChange={(event) => setValue(event.target.value)} />
        <label htmlFor="date">Data da Compra</label>
        <input name="date" type="date" value={date} onChange={(event) => setDate(event.target.value)} />
        <button type="submit">Adicionar</button>
      </form>
      {
        errorMessage && <span>Todos os campos devem ser preenchidos!</span>
      }
    </section>
  )
}

export default InvestimentForm;