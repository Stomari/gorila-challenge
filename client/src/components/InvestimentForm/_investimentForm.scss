.investiment-form {
  @include displayLocal(center);

  form {
    border: 2px solid black;
    border-radius: 15px;
    padding: 20px 20px 20px 20px;

    .form-title {
      h4 {
        margin-top: 0;
        font-size: 1.2rem;
      }
    }

    .form-inputs {
      display: flex;

      .inputs {
        padding-right: 20px;
      }
      button {
        background-image: linear-gradient(to right, $bg-blue, $bg-green);
        border: none;
        border-radius: 20px;
        font-size: 1.2rem;
        padding: 10px;
        cursor: pointer;
        &:focus {
          outline: none;
        }
        &:hover {
          background-image: linear-gradient(to right, $bg-green, $bg-blue);
        }
      }
    }

    .error {
      font-weight: bold;
      color: red;
    }
  }
}

@media (max-width: 1000px) {
  .investiment-form {
    .form-inputs {
      flex-direction: column;
      .inputs {
        padding-bottom: 10px;
      }
    }
  }
}
