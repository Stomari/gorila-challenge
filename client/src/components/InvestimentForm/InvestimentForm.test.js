import React from 'react';
import { shallow } from 'enzyme';
import InvestimentForm from './InvestimentForm';

describe('InvestimentForm', () => {
  const component = shallow(<InvestimentForm />);

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render form', () => {
    expect(component.find('form').length).toEqual(1);
  });

  it('form should have onSubmit props', () => {
    expect(component.find('form').prop('onSubmit')).toEqual(expect.any(Function));
  });

  it('should render type selector', () => {
    expect(component.find('select[name="type"]').length).toEqual(1);
  });

  it('should render value input', () => {
    expect(component.find('input[name="value"]').length).toEqual(1);
  });

  it('should render date input', () => {
    expect(component.find('input[name="date"]').length).toEqual(1);
  });

  it('should render submit button', () => {
    expect(component.find('button[type="submit"]').length).toEqual(1);
  });
});
