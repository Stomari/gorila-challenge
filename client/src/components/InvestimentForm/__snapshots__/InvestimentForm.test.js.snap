// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestimentForm should render 1`] = `ShallowWrapper {}`;
